### PR TITLE
EUI-3587: Fix "Print URL" pipe for rewriting remote Print Service URLs

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3587

### Change description ###
Fix the pipe to rewrite remote Print Service URLs to the correct "local" version to be used for the front-end, irrespective of environment-based variations of the remote domain name.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: demo/src/app/app.config.ts [hunk 1]
```diff
@@ -116,10 +115,6 @@ export class AppConfig extends AbstractAppConfig {
     return this.config.print_service_url;
   }
 
-  public getRemotePrintServiceUrl() {
```
Comment (paul-graham) [inline demo/src/app/app.config.ts:119]: I think this makes this a breaking change and so it would need to go to the next major release...

**Review by paul-graham — CHANGES_REQUESTED** [review]
I think this needs to be a major release: `3.0.0`.

**Review by adnanakgun — APPROVED** [review]
I agree that this is a breaking change. Approving subject to @paul-graham 's change request.

File: demo/src/app/app.config.ts (hunk 1 above)
Comment (Daniel-Lam) [reply demo/src/app/app.config.ts:119]: I've now re-introduced this in `app.config.ts` as a deprecated, no-op function that returns `undefined`, ahead of its removal in the next major release.

File: src/app.config.ts [hunk 2]
```diff
@@ -47,7 +54,8 @@ export class CaseEditorConfig {
   activity_url: string;
   activity_max_request_per_batch: number;
   print_service_url: string;
-  remote_print_service_url: string;
+  // remote_print_service_url marked as optional since deprecation, ahead of removal in next major release
```
Comment (paul-graham) [inline src/app.config.ts:57]:
```
/**
 * @deprecated?
 */
```

Comment (Daniel-Lam) [reply src/app.config.ts]: Good point - done

**Review by paul-graham — APPROVED** [review]
Thank you for making the changes necessary to stop this being a breaking change, @Daniel-Lam. As you've pointed out, version `3.0.0` is earmarked for much more exciting things and this would have put an unnecessary obstacle in the way of that.

Given the deprecation flags and the other changes made, happy to approve this. 👍

**Review by paul-graham — APPROVED** [review]
Changes to the `package.json` and `RELEASE-NOTES.md` look all good. Re-approving. 👍

**Review by adnanakgun — APPROVED** [review]

**Pull request merged**